### PR TITLE
Module Selector

### PR DESCRIPTION
**Pull request by pfitzseb**
Builds on #1231.

### CODE REVIEW

**Comment by pfitzseb** [comment]
Ok, I think this is almost feature complete and adds
- A status bar item that displays the current module and whether it is actually loaded (not loaded when in parens):
image
- A module selector:
image

No automatic detection will happen for the current session and file if you set a module with the module selector (until you select `Choose Automatically).

This isn't at all hooked up to terminals yet. The current module is also not used anywhere, but https://github.com/julia-vscode/julia-vscode/pull/1222 will make use of it.

File: scripts/terminalserver/terminalserver.jl [hunk 1]
```diff
@@ -35,6 +35,56 @@ function sendDisplayMsg(kind, data)
...
+    end
+
+    for m in Symbol.(ms)
+        if isdefined(out, m)
+            resolved = getfield(out, m)
+
+            if resolved isa Module
+                out = resolved
+            else
+                return out
```
Comment (aviatesk) [inline scripts/terminalserver/terminalserver.jl:61]: maybe you want 
```suggestion
                return out !== Main
```
?

File: scripts/terminalserver/terminalserver.jl [hunk 2]
```diff
@@ -35,6 +35,56 @@ function sendDisplayMsg(kind, data)
...
+            end
+        end
+    end
+
+    return out !== Main
+end
```
Comment (aviatesk) [inline scripts/terminalserver/terminalserver.jl:67]: How about refactor this function like
```julia
is_module_loaded(mod) = mod == "Main" || module_from_string(mod) !== Main
```
?

Comment (aviatesk) [reply scripts/terminalserver/terminalserver.jl:67]: nvm, I missed that this branch is on #1231

File: scripts/terminalserver/terminalserver.jl (hunk 1 above)
Comment (pfitzseb) [reply scripts/terminalserver/terminalserver.jl]: Yeah, good catch.

**Comment by davidanthoff** [comment]
Can we delete this branch?

**Comment by pfitzseb** [comment]
Unless someone wants to review this separately, sure.

**Comment by davidanthoff** [comment]
It actually doesn't show any diffs in this PR anymore, for some reason :) I'll delete the branch and we can review as part of the bigger PR.

**Pull request merged**